Datastore/src/Operation.php

```diff
@@ -871,4 +961,275 @@ private function sortEntities(array $entities, array $keys)
 
         return $ret;
     }
+
+    /**
+     * Convert a list of keys to a list of {@see Google\Cloud\Datastore\V1\Key}.
+     *
+     * @param array[] $keys
+     * @return Key[]
+     */
+    private function keysList(array $keys)
+    {
+        $out = [];
+        foreach ($keys as $key) {
+            $local = [];
+
+            if (isset($key['partitionId'])) {
+                $p = $this->arrayFilterRemoveNull([
+                    'project_id' => isset($key['partitionId']['projectId'])
+                        ? $key['partitionId']['projectId']
+                        : null,
+                    'namespace_id' => isset($key['partitionId']['namespaceId'])
+                        ? $key['partitionId']['namespaceId']
+                        : null,
+                    'database_id' => isset($key['partitionId']['databaseId'])
+                        ? $key['partitionId']['databaseId']
+                        : null,
```

Comment:
Use the null coalescing operator for readability
```suggestion
                    'project_id' => $key['partitionId']['projectId'] ?? null,
                    'namespace_id' => $key['partitionId']['namespaceId'] ?? null,
                    'database_id' => $key['partitionId']['databaseId'] ?? null,
```